design/improve_error_handling_todo_list.md

```diff
@@ -0,0 +1,218 @@
+      except Exception as exception:
+          # pylint: disable=line-too-long
+          raise RuntimeError("Unable to authenticate with Conjur. Please log in and try again.") from exception
+      ```
+
+      About the `except Exception`  there we need to check which exception can be thrown . load functino can throw NotLogedInException (written in these document)
+
+* credentials_from_file.py
+
+  * load function
+
+    * NotLogedInException
+
+      ```python
+      if netrc_obj.hosts == {}:
+          raise Exception("You are already logged out")
+      ```
+
+      ```python
+      if netrc_auth == "":
+          raise Exception("You are already logged out")
+      ```
+
+* http_wrapper.py
+
+  * enable_http_logging
+
+* Init_controller.py
+
+  * get_server_certificate function
+
+    * EmptyParameterError
+
+      ```python
+      if self.conjurrc_data.appliance_url == '':
+          # pylint: disable=raise-missing-from
+          raise RuntimeError("Error: URL is required")
+      ```
+
+    * InvalidURLError
+
+      ```python
+      if url.scheme != 'https':
+          raise RuntimeError(f"Error: undefined behavior. Reason: The Conjur URL format provided "
+                 f"'{self.conjurrc_data.appliance_url}' is not supported.")
+      ```
+
+    * UserRefusedException - User dont trust certificate
+
+      ```python
+      trust_certificate = input("Trust this certificate? (Default=no): ").strip()
+      if trust_certificate.lower() != 'yes':
+          raise RuntimeError("You decided not to trust the certificate")
+      ```
+
+  * get_account_info function
+
+    * EmptyParameterError Account
+
+      ```python
+      try:
+          self.init_logic.fetch_account_from_server(self.conjurrc_data)
+      # pylint: disable=broad-except,logging-fstring-interpolation
+      except Exception as error:
+          # pylint: disable=line-too-long,logging-fstring-interpolation
+          logging.warning(f"Unable to fetch the account from the Conjur server. Reason: {error}")
+          # If there was a problem fetching the account from the server, we will request one
+          conjurrc_data.account = input("Enter the Conjur account name (required): ").strip()
+      
+          if conjurrc_data.account is None or conjurrc_data.account == '':
+              # pylint: disable=raise-missing-from
+              raise RuntimeError("Error: account is required")
+      ```
+
+  * __ensure_overwrite_file function
+
+    * UserRefusedException - User refused to overwrite the file
+
+      ```python
+      force_overwrite = input(f"File {config_file} exists. " \
+                              f"Overwrite? (Default=yes): ").strip()
+      if force_overwrite != '' and force_overwrite.lower() != 'yes':
+          raise Exception(f"Not overwriting {config_file}")
+      ```
+
+* Init_logic.py
+
+  * get_certificate function
+
+    * FailedToRetriveCertificateError - On any case SSL service fails.
```

Comment:
`FailedToRetrieveCertificateError`

Reply:
do you think we need more granularity here for certificate errors?

Reply:
Changed to FailedToRetrieveCertificateError